stress/src/throughput.rs

```diff
@@ -58,91 +58,111 @@ where
-                            process.virtual_memory() as f64 / (1024.0 * 1024.0)
-                        );
-                    } else {
-                        println!("Process not found");
+    let shared_mutable_stats_slice = UnsafeSlice::new(&mut worker_stats_vec);
+
+    thread::scope(|s| {
+        s.spawn(|| {
+            let mut last_collect_time = Instant::now();
+            let mut total_count_old: u64 = 0;
+
+            #[cfg(feature = "stats")]
+            let pid = Pid::from(std::process::id() as usize);
+            #[cfg(feature = "stats")]
+            let mut system = System::new_all();
+
+            loop {
+                let current_time = Instant::now();
+                let elapsed = current_time.duration_since(last_collect_time).as_secs();
+                if elapsed >= SLIDING_WINDOW_SIZE {
+                    let total_count_u64 = shared_mutable_stats_slice.sum();
+                    last_collect_time = Instant::now();
+                    let current_count = total_count_u64 - total_count_old;
+                    total_count_old = total_count_u64;
+                    let throughput = current_count / elapsed;
+                    println!(
+                        "Throughput: {} iterations/sec",
+                        throughput.to_formatted_string(&Locale::en)
+                    );
+
+                    #[cfg(feature = "stats")]
+                    {
+                        system.refresh_all();
+                        if let Some(process) = system.process(pid) {
+                            println!(
+                                "Memory usage: {:.2} MB",
+                                process.memory() as f64 / (1024.0 * 1024.0)
+                            );
+                            println!("CPU usage: {}%", process.cpu_usage() / num_threads as f32);
+                            println!(
+                                "Virtual memory usage: {:.2} MB",
+                                process.virtual_memory() as f64 / (1024.0 * 1024.0)
+                            );
+                        } else {
+                            println!("Process not found");
+                        }
                     }
+
+                    println!("\n");
                 }
 
-                println!("\n");
-            }
+                if STOP.load(Ordering::SeqCst) {
+                    break;
+                }
 
-            if STOP.load(Ordering::SeqCst) {
-                break;
+                thread::sleep(Duration::from_millis(5000));
             }
+        });
 
-            thread::sleep(Duration::from_millis(5000));
+        for thread_index in 0..num_threads {
+            let func_arc_clone = Arc::clone(&func_arc);
+            s.spawn(move || loop {
+                func_arc_clone();
+                unsafe {
+                    shared_mutable_stats_slice.increment(thread_index);
+                }
+                if STOP.load(Ordering::SeqCst) {
```

Comment:
Won't we be now forced to check this AtomicBool for each iteration? I think that still be avoided and instead can be checked every BATCH_SIZE?

Reply:
Since the `STOP` is only ever changed when exiting the stress test, the cache line where `STOP` resides wouldn't have to be updated during the test. So, I don't expect it to have much of a perf implication. (don't think it suffers from false sharing here as `STOP` is `static`).

If we still want to avoid it, I'd prefer if remove the `STOP` variable altogether and simply exit the process on `Ctrl + C`. The existing code for using `STOP` doesn't add much value.

Reply:
If we still want to avoid it, I'd prefer if remove the `STOP` variable altogether and simply exit the process on `Ctrl + C`.

I ran the stress test with the above-mentioned change, and there isn't much of a difference in the results. I would prefer removing it as the existing code for using `STOP` doesn't add much value.

Reply:
not a blocker for this PR, lets revisit removing it alltogether in a future pr, if needed.